QuizApp/Quiz/ViewControllers/QuizViewController.swift

```diff
@@ -178,9 +167,69 @@ extension QuizViewController {
                 let section = NSCollectionLayoutSection(group: group)
                 section.contentInsets = NSDirectionalEdgeInsets(top: 10, leading: 10, bottom: 10, trailing: 10)
                 section.interGroupSpacing = 10
+                let headerFooterSize = NSCollectionLayoutSize(
+                    widthDimension: .fractionalWidth(1.0),
+                    heightDimension: .estimated(20))
+                let sectionHeader = NSCollectionLayoutBoundarySupplementaryItem(
+                    layoutSize: headerFooterSize,
+                    elementKind: UICollectionView.elementKindSectionHeader,
+                    alignment: .top)
+                section.boundarySupplementaryItems = [sectionHeader]
                 return section
             })
         return layout
     }
 
+    func applySnapshot(allQuizzes: Bool = false, animatingDifferences: Bool = true) {
```

Comment:
`applySnapshot` should get quiz models injected as parameters and then ideally append sections and items only based on the received models.  Basically, if quizViewModel.quizzes contains all sections then you would append all sections, if it contains just movies, then that is what you need to append. But that needs to be resolved based on the content of quizViewModel.quizzes, not based on selected category.